packages/server/src/users/users.controller.ts

```diff
@@ -1,7 +1,76 @@
-import { Controller } from '@nestjs/common';
+import { Body, Controller, Delete, Get, Put, Query, UseGuards } from '@nestjs/common';
+import { ApiOperation, ApiSecurity, ApiTags } from '@nestjs/swagger';
+import { JWT_SECURITY } from '../auth/auth.constants';
+import { JwtAuthGuard } from '../auth/guards/jwt.guard';
+import { RolesGuard } from '../auth/guards/role.guard';
+import { CurrentUser } from '../common/decorators/current-user.decorator';
+import { Roles } from '../common/decorators/user-role.decorator';
+import { UpdateCriteriaDto } from './dto/update-criteria.dto';
+import { User, UserRole } from './entities/user.entity';
 import { UsersService } from './users.service';
 
 @Controller('users')
+@ApiTags('Users')
+@UseGuards(JwtAuthGuard, RolesGuard)
 export class UsersController {
   constructor(private readonly usersService: UsersService) {}
+
+  @Get('/criteria')
+  @ApiSecurity(JWT_SECURITY)
+  @ApiOperation({
+    summary: 'Get notification criteria for a user',
+    description:
+      'Get the critera (councils/meeting updates) that determines when a user will receive a notification',
+  })
+  async getUserNotificationCriteria(@CurrentUser() user: User): Promise<string[]> {
+    const usr = await this.usersService.getUser(user.email, true);
+    return usr.councils;
+  }
+
+  @Put('/criteria')
+  @ApiSecurity(JWT_SECURITY)
+  @ApiOperation({
+    summary: 'Update notification criteria for a user',
+    description:
+      'Set the critera (councils/meeting updates) that determines when a user will receive a notification',
+  })
+  updateUserNotificationCriteria(
+    @CurrentUser() user: User,
+    @Body() criteria: UpdateCriteriaDto,
+  ): Promise<User> {
+    return this.usersService.updateUserNotificationCriteria(user, criteria);
+  }
+
+  @Delete('/me')
+  @ApiSecurity(JWT_SECURITY)
+  @Roles(UserRole.USER)
+  @ApiOperation({
+    summary: 'Delete current user',
+    description: 'Allows a user to delete themselves',
+  })
+  deleteSelf(@CurrentUser() user: User): Promise<User> {
+    return this.usersService.deleteUser(user.email);
+  }
```

Comment:
No need for a dedicated endpoint for users to delete themselves. They'll have access to their internal ID and should just be able to pass that to the endpoint.